### ISSUE TITLE
AudioSwitcher freezes at launch in Win10

### ISSUE THREAD

**Issue by sambul13** [xenolightning/AudioSwitcher_v1#2173]
Testing AudioSwitcher 1.8.0.42 in Windows 10 version 1709 with latest updates. At first launch AudioSwitcher GUI freezes indefinitely, and ProcessExplorer shows one svchost.exe process using 25% CPU load related to Windows Audio. If I manually restart the svchost process, it completes normally, CPU load drops to normal, and AudioSwitcher GUI unfreezes without restart. When closed and then reopen, AudioSwitcher seems to work normally, at times from 2nd or 3rd launch.

**Comment by xenolightning** [comment]
I've seen this only once before.

I think it may be related to the number of audio devices on the system/

How many audio devices are available, in the control panel menu? Including disabled/disconnected?

**Comment by sambul13** [comment]
Its repeatable after each PC reboot and AudioSwitcher launch. Eventually Windows Audio Troubleshooter kicks in, resets the provider, and AudioSwitcher launches. Historically there're a dozen of Speaker devices registered in the system, and about the same number of Mics, but only a few are currently enabled, the rest is disabled.

This package is the only one among those I tried causing such issue. I like it as I can assign *any* shortcuts to devices, but it needs fixing. For example, [SoundSwitch ](https://github.com/Belphemur/SoundSwitch)launches instantly on the same system.

**Comment by d3xt3r01** [comment]
I can confirm.

In playback I have these:
```
Realtek Digital Output - Realtek High Definition Audio
Realtek HD Audio 2nd output - Realtek High Definition Audio
Speakers - Sound Blaster Z
SPDIF-Out - Sound Blaster Z
DELL P2715Q - NVIDIA High Definition Audio
Speakers - BLUE Yeti PRO
Speakers - 4- Dell AC511 USB Soundbar
DELL P2715Q - NVIDIA High Definition Audio
```

In recording I have these:
```
What U Hear - Sound Blaster Z
Line - 4- Dell AC511 USB SoundBar
Microphone - Sound Blaster Z
Digital-In - Sound Blaster Z
Microphone - BLUE Yeti PRO
Microphone - HD Pro Webcam C920
```

**Comment by recon88** [comment]
Same issue after latest Windows 10 update. Was using it just fine until today for several months.

**Comment by xenolightning** [comment]
@recon88 Ahh!

What's the version of Windows? 

I think I did get part way into figuring out this issue, but it was intermittent at best on my test system.

Maybe with the new version I can reliably re-produce it.

**Comment by recon88** [comment]
I had to launch it, then double left-click the icon and spam left-click the interface until it started responding. Now it's working fine for a couple days in a row.

**Comment by PlayingKarrde** [comment]
Just updated to Windows 1803 and now getting this problem also. Spamming left clock as @xenolightning suggested did work though.

**Comment by Devocub** [comment]
Bump. 1 core full load by "Windows Audio" service, audioswitcher is frozen during this time (not even fully loaded because right click on icon in tray shows default layout). ~30 sec and it's loaded. Killing audioswitcher process instantly fix this cpu load. Without audioswitcher in startup there's no issue. 2nd+ restarts of audioswitcher doesn't cause cpu load.

**Comment by xenolightning** [comment]
I think Windows updates is causing issues with devices, and the startup process must be deadlocking audio switcher for some reason. It's in the audio session/volume code somewhere, just need to find what is doing it.

I'm working on fixing the underlying issues with the sound system, and am hoping to have this fixed (eventually in v2) but applying it to v1 first for a wider user base

**Comment by andycjw** [comment]
where to get this fix for latest windows 10?

**Comment by dryocampa** [comment]
This issue also started popping up for me after upgrading from Windows 7 to 10, unfortunately. 

The locked svchost tends to also break all sound input/output on my system while ongoing as well, the effect lasting longer than any other startup item by several minutes. 

Seems development is on the backburner for now, but I'd love a fix in the future!

**Comment by recon88** [comment]
So any news on that? It has been almost a year since your last reply

**Comment by Devocub** [comment]
seem to be was windows bug, rn i don't notice any problems (on latest windows)

**Comment by xenolightning** [comment]
@Devocub interesting - what build of Windows 10 is it? It did seem that a change in Windows 10 caused instability, but could never prove it as such.

**Comment by Devocub** [comment]
@xenolightning latest

**Comment by Devocub** [comment]
ok i found something
if you ever had relatek audio drivers installed then it will load way longer and strain cpu core
@xenolightning

**Comment by UrbanPotato** [comment]
yea this seems to be an issue for me as well disabling some unused devices does seem to help some but it doesn't completely mitigate it

**Comment by superblobster** [comment]
WHEN will you fix this damn bug??? It's been over a year, what are you doing? I'm going to have to switch (haha) to a better software since this keeps happening almost every time.

**Comment by PlayingKarrde** [comment]
I moved to nircmd setup as a hotkey on my keyboard using autohotkey and havent looked back since. I definitely recommend this method. Perfectly reliable.

**Comment by dryocampa** [comment]
Belphemur's SoundSwitch (mentioned earlier in thread) has identical practical function to AudioSwitcher, and does not have this issue.

I recommend trying out SoundSwitch if you're experiencing this bug. Changing software is obviously not an ideal solution, but it's fairly painless, and I made the switch ~9 months ago and have been perfectly satisfied

https://github.com/Belphemur/SoundSwitch

**Comment by superblobster** [comment]
dryocampa, thank you so much! SoundSwitch is perfect. Haven't had any problems or high CPU usage so far.

Good riddance, AudioSwitcher! I hardly knew you.

**Comment by Luckz** [comment]
Perhaps there's an issue with specific Realtek, Nvidia or other audio drivers? When I set up this system I installed some older specific Realtek version (realtek-6.0.1.8459-20180605) to avoid issues I had just using the newest / default ones.
Of course it could also just be the sheer amount of devices. I'm only at 4 non-disabled playback devices and 3 non-disabled recording devices, while one poster above listed 14 devices.

**Comment by superblobster** [comment]
> Perhaps there's an issue with specific Realtek, Nvidia or other audio drivers? When I set up this system I installed some older specific Realtek version (realtek-6.0.1.8459-20180605) to avoid issues I had just using the newest / default ones.
> Of course it could also just be the sheer amount of devices. I'm only at 4 non-disabled playback devices and 3 non-disabled recording devices, while one poster above listed 14 devices.

No, I don't think so. I have the newest standard drivers from Realtek and Nvidia, and only 3 devices.

**Comment by dryocampa** [comment]
You're very welcome superblobster! Glad I could help. 

Unfortunately, the final answer here is that this repository has not been actively maintained since 2018, and as the project name suggests, we're trying to find a fix for obsolete software with several functioning alternatives that evolved into .NET libraries and APIs a good while back. 

It sucks, but if you're experiencing this bug, just try out the very actively maintained SoundSwitch linked above and save yourself the hassle. Thread (unofficially) closed

**Comment by AMBIztw** [comment]
> Testing AudioSwitcher 1.8.0.42 in Windows 10 version 1709 with latest updates. At first launch AudioSwitcher GUI freezes indefinitely, and ProcessExplorer shows one svchost.exe process using 25% CPU load related to Windows Audio. If I manually restart the svchost process, it completes normally, CPU load drops to normal, and AudioSwitcher GUI unfreezes without restart. When closed and then reopen, AudioSwitcher seems to work normally, at times from 2nd or 3rd launch.

Bump. I really love Audio Switcher but this bug annoys. Man @xenolightning please fix it.

**Comment by vertigo220** [comment]
Just tried this after giving up on SoundSwitch due to its own issues, and both times I've tried it, it has been basically unresponsive, says there's an update available despite having the latest version, and I have to kill the process, which takes a bit of time. This is on the latest version of Win10. Looks like this is dead software, unfortunately.

**Comment by Luckz** [comment]
> Just tried this after giving up on SoundSwitch due to its own issues, and both times I've tried it, it has been basically unresponsive, says there's an update available despite having the latest version, and I have to kill the process, which takes a bit of time. This is on the latest version of Win10. Looks like this is dead software, unfortunately.

Disable the update check?

I use this on any desktop computer.

Instead of downloading from GH, you can also obtain it through Scoop or WinGet.

**Comment by vertigo220** [comment]
Doesn't matter how you get it, it's going to be the same, and it's unusable since it freezes immediately, and not only should the update check not need to be disabled, but it's not exactly an option given the app is unresponsive.

**Comment by Luckz** [comment]
> Doesn't matter how you get it, it's going to be the same, and it's unusable since it freezes immediately, and not only should the update check not need to be disabled, but it's not exactly an option given the app is unresponsive.

%localappdata%\AudioSwitcher\AudioSwitcher.json

```
{
...
   "PollForUpdates" : "0",
...
   "UpdateNotificationsEnabled" : "False"
}
```

Prolly one of those two?

**Comment by vertigo220** [comment]
That would probably work, but I fail to see the point when the app is clearly broken. Even if I were to disable update checks, the fact it's showing there's an update when there isn't shows that it has major issues, not to mention the fact it's still unusable since it's unresponsive. The update issue, while a definite bug, is the lesser issue here.
